packages/core/src/libs/patronus/postpone.ts

```diff
@@ -12,9 +22,39 @@ export function postpone<T>({
   clock: Event<T>;
   until: Store<boolean>;
 }): EventAsReturnType<T> {
-  return sample({
-    clock: [clock, until],
-    source: clock,
+  const target = createEvent<T>();
+
+  const $planned = createStore({ ref: new Set<T>() }, { serialize: 'ignore' });
```

Comment:
Why use `Set`, if you later convert it to array via `set.values()` anyway? 🤔 

I think, it would be easier to just use array instead?

Reply:
Got rid of this complex solution, now it's way simplier 🤗